aws/hhvm1/worker.sh

```diff
@@ -0,0 +1,158 @@
+  echo "Must set ACTIVITY_ARN and SCRIPT_URL"
+  fail
+fi
+
+# This script needs to be as robust as possible, so we retry just about
+# everything important that could reasonably fail.
+try_really_hard() {
+  for I in $(seq 5); do
+    "$@" && return 0
+    sleep 10
+  done
+  fail
+}
+
+# Using snap because apt-get's version of aws-cli is too old.
+try_really_hard snap install jq
+try_really_hard snap install aws-cli --classic
+
+export PATH="$PATH:/snap/bin"
+
+aws configure set default.region us-west-2
+EC2_INSTANCE_ID=$(curl http://169.254.169.254/latest/meta-data/instance-id)
+
+if [ -z "$SKIP_CLOUDWATCH" ]; then
+  CLOUDWATCH_CONFIG_FILE="$(mktemp)"
+  cat > "${CLOUDWATCH_CONFIG_FILE}" <<EOF
+[general]
+state_file = /var/awslogs/state/agent-state
+
+[/var/log/cloud-init-output.log]
+file = /var/log/cloud-init-output.log
+log_group_name = hhvm-binary-package-builds/cloud-init-output.log
+log_stream_name = $(date "+%Y/%m/%d")/w_$EC2_INSTANCE_ID
+EOF
+  curl --retry 5 -O \
+    https://s3.amazonaws.com//aws-cloudwatch/downloads/latest/awslogs-agent-setup.py
+  python3 awslogs-agent-setup.py -n -r us-west-2 -c "${CLOUDWATCH_CONFIG_FILE}"
+fi
+
+curl --retry 5 "$SCRIPT_URL" > process-task.sh || fail
+
+if [ -n "$INIT_URL" ]; then
+  curl --retry 5 "$INIT_URL" > init.sh || fail
+  source init.sh
+fi
+
+PROCESSED_COUNT=0
+
+# Each poll is ~1 minute, so this is the approximate number of minutes we'll
+# wait for the next task before shutting down.
+MAX_POLLS=10
+
+# Cancel the hardcoded shutdown (the main loop below has more nuanced shutdown
+# logic of its own).
+shutdown -c
+
+while true; do
+  for I in $(seq $MAX_POLLS); do
+    echo "Waiting for the next task ($I/$MAX_POLLS)..."
+
+    TASK_JSON=$(
+      aws stepfunctions get-activity-task \
+        --activity-arn "$ACTIVITY_ARN" \
+        --worker-name "ec2_$EC2_INSTANCE_ID" \
+        --cli-read-timeout 70 \
+        --output json
+    )
+    TASK_TOKEN=$(echo "$TASK_JSON" | jq -r '.taskToken // ""')
+
+    if [ -n "$TASK_TOKEN" ]; then
+      PROCESSED_COUNT=$((PROCESSED_COUNT + 1))
+
+      # We got a task, run it!
+      INPUT_JSON=$(echo "$TASK_JSON" | jq -r '.input // {}')
+      TASK_NAME=$(echo "$INPUT_JSON" | jq -r '.name // ""')
+      TASK_ENV=$(echo "$INPUT_JSON" | jq -r '.env // ""')
+
+      # Update EC2 instance name to reflect the name of the current task (for
+      # nicer view in the AWS console).
+      if [ -n "$EC2_INSTANCE_ID" -a -n "$TASK_NAME" ]; then
+        aws ec2 create-tags --resources "$EC2_INSTANCE_ID" \
+          --tags "Key=Name,Value=wa-$PROCESSED_COUNT-$TASK_NAME"
+      fi
+
+      # Heartbeats are not used right now but could be useful at some point.
+      # echo "*/5 * * * * aws stepfunctions send-task-heartbeat '$TASK_TOKEN'" \
+      #   | crontab -
+
+      WRAPPER_SCRIPT="./task_${TASK_NAME}_$(date +%Y-%m-%d_%H-%M-%S).sh"
+      echo "#!/bin/bash
+        shutdown() {
+          echo skipping shutdown \$@
+        }
+        SKIP_CLOUDWATCH=1
```

Comment:
these are temporary hacks so that I can reuse the same scripts between the old and new flow, can be cleand up later